pkg/patterns/declarative/reconciler.go

```diff
@@ -170,26 +175,60 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		}
 	}()
 
+	original := instance.DeepCopyObject().(DeclarativeObject)
+
 	if r.options.status != nil {
 		if err := r.options.status.Preflight(ctx, instance); err != nil {
 			log.Error(err, "preflight check failed, not reconciling")
 			return reconcile.Result{}, err
 		}
 	}
 
-	objects, err = r.reconcileExists(ctx, request.NamespacedName, instance)
+	var statusInfo *StatusInfo
+	statusInfo, err = r.reconcileExists(ctx, request.NamespacedName, instance)
+	objects = statusInfo.Manifest // for the defer block
+	if err != nil {
+		statusInfo.Err = err
+	}
 
 	if err != nil {
 		r.recorder.Eventf(instance, "Warning", "InternalError", "internal error: %v", err)
 	}
 
+	if r.options.status != nil {
+		if err := r.options.status.BuildStatus(ctx, statusInfo); err != nil {
+			log.Error(err, "error building status")
+			return result, err
+		}
+		oldStatus, err := utils.GetCommonStatus(original)
+		if err != nil {
+			log.Error(err, "error getting status")
+			return result, err
+		}
+		newStatus, err := utils.GetCommonStatus(instance)
+		if err != nil {
+			log.Error(err, "error getting status")
+			return result, err
+		}
+		if !reflect.DeepEqual(oldStatus, newStatus) {
+			if err := r.client.Status().Update(ctx, instance); err != nil {
```

Comment:
Q: How would the `Status().Update(..)` resolves update conflict when the user is modifying the value in the CR?

Reply:
So the user doesn't normally modify the status.  But in any case, the call to Update actually uses optimistic concurrency.  We provide the same `instance` that we read.  That instance contains a ResourceVersion.  If there was a concurrent change, the resourceVersion on the server does not match, and the Update is rejected.  That is returned as an error, we return an error from the reconciler, and the reconciler is run again (with backoff).

It doesn't normally happen because this controller _should_ (generally) own the Status.  So we don't handle it specially.

Reply:
Thank you for the explanation.